www/src/context/InsightsContext.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import * as Components from '~paragon-react'; // eslint-disable-line
+import * as Icons from '~paragon-icons'; // eslint-disable-line
+import getParagonComponentsTypes from '../utils/getParagonComponentsTypes';
+
+const InsightsContext = createContext({});
+
+const InsightsContextProvider = ({ children }) => {
+  const [paragonTypes, setParagonTypes] = useState({});
+
+  useEffect(() => {
+    setParagonTypes(getParagonComponentsTypes(Components));
+  }, []);
+
+  const isParagonIcon = (name) => name in Icons;
+
+  const contextValue = {
+    paragonTypes,
+    isParagonIcon,
+  };
```

Comment:
Generally, context values should be memoized (e.g., with `useMemo`) to avoid re-renders. Each re-render of `InsightsContextProvider` would cause `contextValue` to get re-created, which will ultimately causes any children of `InsightsContext.Provider` to re-render as well.

Reply:
Wrapped in `useMemo`. It will be triggered once when `useEffect` changes value for `paragonTypes`.